clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -289,12 +289,16 @@ private long nextOffset(TopicPartition tp) {
         }
     }
 
-    public synchronized void flush() {
+    public synchronized void flush(Duration timeout) {
         verifyProducerState();
```

Comment:
do we want to maybe respect the timeout? (even though this is a mock)

Reply:
@kidkun can you respond to/resolve this comment?

E.g., we could just throw an `UnsupportedOperationException` if `timeout != Long.MAX_VALUE`?